docs/input-kafka.asciidoc

```diff
@@ -344,6 +345,27 @@ NOTE: In cases when multiple inputs are being used in a single pipeline, reading
 it's essential to set a different `group_id => ...` for each input. Setting a unique `client_id => ...`
 is also recommended.
 
+[id="plugins-{type}s-{plugin}-group_instance_id"]
+===== `group_instance_id`
+
+* Value type is <<string,string>>
+* Optional, there is no default value for this setting.
+
+The static membership identifier for this consumer. Static membership feature was introduced in
+https://cwiki.apache.org/confluence/display/KAFKA/KIP-345%3A+Introduce+static+membership+protocol+to+reduce+consumer+rebalances[KIP-345],
+available under Kafka property `group.instance.id` and aims to avoid rebalances in contexts where a lot of data
+has to be handed off after a consumer goes offline.
+This mitigates the cases where the service state is heavy and rebalance of one topic partition from instance
+A to B means huge amount of data transfer. Client that goes offline and online frequently, if using this settings
+avoid frequent and heavy rebalances.
```

Comment:
```suggestion
This feature mitigates cases where the service state is heavy and the rebalance of one topic partition from instance
A to B would cause a huge amount of data to be transferred.
A client that goes offline/online frequently can avoid frequent and heavy rebalances by using this option.
```